sdk/eventhub/event-hubs/src/eventHubConsumerClientModels.ts

```diff
@@ -0,0 +1,41 @@
+
+import { CloseReason, PartitionContext, EventProcessorCommonOptions } from './eventProcessor';
+
+/**
+ * Optional event handlers that provide more context when subscribing to events.
+ */
+export interface OptionalEventHandlers {
+  onError?: (error: Error, context: PartitionContext) => Promise<void>;
+  onInitialize?: (context: PartitionContext) => Promise<void>;
+  onClose?: (reason: CloseReason, context: PartitionContext) => Promise<void>;
+
+}
+
+/**
+ * Options for subscribe.
+ */
+export interface SubscriptionOptions extends OptionalEventHandlers, EventProcessorCommonOptions {
+}
+
+/**
+ * Represents the status of a subscribe() call and can be used to stop a subscription.
+ */
+export interface Subscription {
+  /**
+   * Stops the subscription from receiving more messages.
+   * @returns Promise<void>
+   * @throws {Error} Thrown if the underlying connection encounters an error while closing.
+   */
+  stop(): Promise<void>;
+  /**
+   * @property Indicates whether the receiver is connected/open.
+   * `true` - is open; `false` otherwise.
+   * @readonly
+   */
+  isReceiverOpen(): boolean;
+  /**
+   * @property The consumer group from which the handler is receiving events.
+   * @readonly
+   */
+  consumerGroup(): string;
```

Comment:
The reason I thought this wouldn't be needed on the subscription is because it should be passed to each handler invocation via the partitionContext.